src/ripple/basics/impl/ProgressDisplay.cpp

```diff
@@ -0,0 +1,105 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2020 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/basics/ProgressDisplay.h>
+#include <iomanip>
+#include <iostream>
+
+namespace ripple {
+
+ProgressDisplay::ProgressDisplay(
+    std::uint64_t const expectedCount,
+    std::string const& countedItemName,
+    std::string const& initialMessage)
+    : expectedCount_(expectedCount)
+    , countedItemName_(countedItemName)
+    , initialMessage_(initialMessage)
+{
+    updateDisplay(0);
+}
+
+ProgressDisplay::~ProgressDisplay()
+{
+    if (itemsCompleted_ < expectedCount_)
+    {
+        std::cout << "Incomplete!" << std::endl;
```

Comment:
> Do we really need a progress bar right now?
> 
> If yes, there's a very popular C++ library for progress bars called [indicators](https://github.com/p-ranav/indicators). Can we use that instead?

The progress bar was requested specifically. Is it truly better to add another library to our repo just to avoid the strangeness of writing to `cout`?

Reply:
@thejohnfreeman We do not need a progress bar. Since this process requires the node operator privilege, I prefer a simple completion percentage returned from the command-line.